openslides_backend/action/actions/user/save_saml_account.py

```diff
@@ -139,18 +144,13 @@ def create_events(self, instance: Dict[str, Any]) -> Iterable[Event]:
         """
         Handles create and update
         """
-        fqid = fqid_from_collection_and_id(self.model.collection, instance["id"])
         if "meta_new" in instance:
-            del instance["meta_new"]
-            yield self.build_event(EventType.Create, fqid, instance)
+            yield from super().create_events(instance)
         else:
-            fields = UpdateAction.create_update_events(instance)
-            if not fields:
-                return []
-            fields = {k: v for k, v in instance.items() if v != self.user.get(k)}
-            if not fields:
-                return []
-            yield self.build_event(EventType.Update, fqid, fields)
+            fields = {
+                k: v for k, v in instance.items() if k == "id" or v != self.user.get(k)
+            }
+            yield from UpdateAction.create_events(cast(UpdateAction, self), fields)
```

Comment:
Now you're emulating a static method without it actually being one, this is really hacky. What I had in mind was that you just copy the 
```
if not fields:
    return []
yield self.build_event(EventType.Update, fqid, fields)
```
which could by the way rewritten to
```
if fields:
    yield self.build_event(EventType.Update, fqid, fields)
```
from the method in the update action to here. Alternatively, if you want to keep the code deduplication, you have to reintroduce the static method or use one of the solutions we discussed (like inheritance, mixin, etc.). But this is not the way to go...

Reply:
Now I'm calling the super method's of the correct base classes, the update and the create one's. There is no hack at all!